core/src/toga/widgets/imageview.py

```diff
@@ -79,6 +81,16 @@ def __init__(
         # Prime the image attribute
         self._image = None
         self._impl = self.factory.ImageView(interface=self)
+        # checking if the image if PIL.Image
+        try:
+            from PIL import Image as _PIL_Image_Module_
+            if _PIL_Image_Module_.isImageType(image):
+                image = Image(pil_image=image) #TODO: A change may be necessary if behaviour of Image is changed
+        except ImportError as e:
+            # cant import... assume that PIL.Image will not be passed
+            pass
```

Comment:
If we're doing this right, there shouldn't be a need for any PIL checks outside the Image class. You can just pass the PIL image directly to Image.